src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinRTWebSocket.cs

```diff
@@ -105,6 +114,20 @@ public override string SubProtocol
                 websocketControl.SupportedProtocols.Add(subProtocol);
             }
 
+            if (MessageWebSocketClientCertificateSupported)
+            {
+                // Use the first app-provided client cert that can be successfully converted into a WinRT client cert (if any).
+                foreach (X509Certificate dotNetClientCert in options.ClientCertificates)
+                {
+                    RTCertificate winRtClientCert = ConvertDotNetClientCertToWinRtClientCert(dotNetClientCert);
+                    if (winRtClientCert != null)
```

Comment:
We throw a lot of exceptions and sometimes differently for different implementations. In this case I would expect that the call fails as we cannot honor the given configuration. Something on the lines of "the client certificate will not be used as we cannot access the private key". We don't need to invent a new exception - we can easily reuse exceptions thrown by ConnectAsync such as WebSocketException, ArgumentException, InvalidOperationException, etc.